### PR TITLE
fix: fail on http error

### PR DESCRIPTION
**Pull request by johanneswuerbach**
Fail when the API returns a non 200 response and show the received http response.

### CODE REVIEW

File: action.test.ts [hunk 1]
```diff
@@ -84,9 +87,13 @@ describe('action', () => {
...
   });
 
+  afterEach(() => {
+    process.exitCode = undefined;
+  });
+
   test('succeeds', async () => {
     await runAction();
-    expect(process.exitCode).toBeFalsy;
+    expect(process.exitCode).toBeFalsy();
```
Comment (johanneswuerbach) [inline action.test.ts:96]: Added `plugin:jest/recommended` to catch this next time.

**Review by delca85 — COMMENTED** [review]
Shouldn't we also push the rebuilt action?

File: action.ts [hunk 2]
```diff
@@ -104,6 +105,13 @@ export async function runAction() {
     core.error('Unable to notify Humanitec about build.');
     core.error('Did you add the token to your Github Secrets? ' +
       'http:/docs.humanitec.com/connecting-your-ci#github-actions');
+
```
[the remark this answers is not in the document]
Comment (johanneswuerbach) [reply action.ts:107]: Updated 👍

**Comment by johanneswuerbach** [comment]
Both resolved ✅

**Review by delca85 — APPROVED** [review]

**Pull request merged**